### DISCUSSION TITLE
[Feature/Trick] Image Slice Rotation

### DISCUSSION THREAD

**Discussion by torridgristle** [eps696/aphantasia#17] [Ideas]
ViT-B/32 appears to only recognize things that are correctly oriented, like a tree won't be crooked and a face won't be upside-down, so adding rotation to the slice_imgs section should result in wider diversity at low levels of random rotation, and something approaching chaos at high levels of random rotation.

Alternatively, I suppose it could be rotated in such a way that the slices all point toward the center of the image, making a radial design converging in the center, rather than rectangular. I reckon that'd work like (offsetx - csize * 0.5) and (offsety - csize * 0.5) to get the center of the slice, and then take those as slicex and slicey for atan2(slicex - centerx, slicey - centery)*(180.0/π) for the angle of rotation for the slice, or rather the angle to rotate the image before slicing and the inverse rotation after slicing.

**Comment by eps696** [comment]
1) i have tested rotations at the very beginning (as a part of standard torch transforms, coming from library) and naturally disliked the effect. sorry to say, neither do i understand the benefit of "radial design" here.  
2) this repo has been created to explore the fundamental features of a novel ML image generation method. feel free to fork this repo for further development of cosmetic usability on your taste.
